### PR TITLE
feat: simplify the import / export API for the `Keybase`

### PR DESCRIPTION
**Pull request by zivkovicmilos**
## Description

This PR greatly simplifies the `Keybase` interface in regards to key imports / exports, by removing redundant methods.

The goal of the PR is to put the application-level key armor management outside the keybase (not including storage), towards the caller. In fact, this is the reason why the `Keybase` has a chaotic API for imports / exports.

I've preserved existing `gnokey` functionality -- this is unchanged.
I will update `gnokey` imports / exports in a separate PR, to make the flag usage make more sense.

**BREAKING CHANGE:**
- Removed 9 `Keybase` methods, and added 2 new ones (that replace them)

cc @jefft0 

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
</details>

### CODE REVIEW

File: tm2/pkg/crypto/keys/types.go
Comment (jefft0) [inline tm2/pkg/crypto/keys/types.go:48]: This comment is still true, but it only applies to `ImportPrivKey` (not to `Rotate`).

Comment (zivkovicmilos) [reply tm2/pkg/crypto/keys/types.go]: Updated:

[21416bd](https://github.com/gnolang/gno/pull/3285/commits/21416bd8687f51fc24806e867e7a51bcde5e922f)

File: tm2/pkg/crypto/keys/client/export.go
Comment (aeddi) [inline tm2/pkg/crypto/keys/client/export.go:109]: This block is duplicated in the if (95-98) and else (106-109) branches. You could place it before the if to deduplicate it.

File: tm2/pkg/crypto/keys/client/import.go
Comment (aeddi) [inline tm2/pkg/crypto/keys/client/import.go:132]: These two blocks are also duplicated (132-141) and (150-159). You could move them after the if-else.

**Review by aeddi — APPROVED** [review]
Two small details, but apart from that, looks good to me 👍

File: tm2/pkg/crypto/keys/types.go
Comment (moul) [inline tm2/pkg/crypto/keys/types.go:55]:
```suggestion
	ExportPrivKey(name string, decryptPass, encryptPass string) (crypto.PrivKey, error)
```

File: tm2/pkg/crypto/keys/types.go
Comment (moul) [inline tm2/pkg/crypto/keys/types.go:52]:
```suggestion
	ImportPrivKey(name string, key crypto.PrivKey, decryptPass, encryptPass string) error
```

**Comment by moul** [comment]
As someone using this package, I prefer this new API. But I would like to summon a few people:  
-  @piux2, as I recall he was using this package for hiring purposes and may have insights that I don't.  
-  @tbruyelle, for his broader experience in other cosmos chains.

**Comment by Gno2D2** [comment]
I'm a bot that assists the Gno Core team in maintaining this repository. My role is to ensure that contributors understand and follow our guidelines, helping to streamline the development process.

The following requirements must be fulfilled before a pull request can be merged.
Some requirement checks are automated and can be verified by the CI, while others need manual verification by a staff member.

These requirements are defined in this [configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

## Automated Checks

 🔴 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🔴 The pull request head branch must be up-to-date with its base ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch))


## Manual Checks

*No manual checks match this pull request.*

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>The pull request head branch must be up-to-date with its base (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 Head branch (dev/zivkovicmilos/keybase-cleanup) is up to date with base (master): behind by 18 / ahead by 6

```
</blockquote></details>

</blockquote></details>



</blockquote></details>

**Review by tbruyelle — COMMENTED** [review]
The SDK equivalent `Keyring` struct exposes a similar API to the actual `Keybase` of this repo (see [1] [2] [3]) (before this PR), without the method that returns the unencrypted private key (namely `ExportPrivateKeyObject()`). This method is not part of the public interface, so it's a bit harder to call (example from the CLI export command: [4]).

However, I see no reason why the API couldn't be simplified. Methods like `Export/ImportPubkey()` are not used anywhere in the SDK, yet they have been around for years.

Now about the "obfuscation" of unsafe `ExportPrivateKeyObject()`, the current change makes a clear difference with the SDK keyring, which only exposes safe methods, while here **only** the unsafe method is exposed. That said, since `ExportPrivateKeyObject()` was already part of the public API, this may not really matter.

[1]: https://github.com/cosmos/cosmos-sdk/blob/main/crypto/keyring/keyring.go#L58
[2]: https://github.com/cosmos/cosmos-sdk/blob/34f407d6367a16b7cc21ac9bded75f1ed8678fc2/crypto/keyring/keyring.go#L121
[3]: https://github.com/cosmos/cosmos-sdk/blob/34f407d6367a16b7cc21ac9bded75f1ed8678fc2/crypto/keyring/keyring.go#L136
[4]: https://github.com/cosmos/cosmos-sdk/blob/34f407d6367a16b7cc21ac9bded75f1ed8678fc2/client/keys/export.go#L83

**Review by kristovatlas — COMMENTED** [review]
I spent about an hour reviewing, and this all looks good to me to improving the interface and code clarity.

File: tm2/pkg/crypto/keys/types.go
Comment (zivkovicmilos) [reply tm2/pkg/crypto/keys/types.go:55]: The intention of this PR was to remove the moment where the keybase API has to manage armor and application-level formats (such as encrypted / unencrypted armor).

Locally, the keybase uses armor to save, but the API should not have this limitation, hence why `encryptPass` was dropped

File: tm2/pkg/crypto/keys/types.go
Comment (zivkovicmilos) [reply tm2/pkg/crypto/keys/types.go:52]: https://github.com/gnolang/gno/pull/3285#discussion_r1890110419

File: tm2/pkg/crypto/keys/client/export.go
Comment (zivkovicmilos) [reply tm2/pkg/crypto/keys/client/export.go]: Good catch 💯 

[3b571b5](https://github.com/gnolang/gno/pull/3285/commits/3b571b5927e6df79e853bea6d0577405ed826521)

File: tm2/pkg/crypto/keys/client/import.go
Comment (zivkovicmilos) [reply tm2/pkg/crypto/keys/client/import.go]: Good catch 💯 

These "features" is something I'll change in a separate PR 🙏 

[3acb4c6](https://github.com/gnolang/gno/pull/3285/commits/3acb4c690b8e302fad52e57a6e04c27b1393d024)

**Pull request merged**
